moveit_planners/ompl/ompl_interface/test/test_constrained_state_validity_checker.cpp

```diff
@@ -0,0 +1,326 @@
+
+    // too much code, just to print the state
+    if (VERBOSE)
+    {
+      // cast ompl state to a specific type, useful in the rest of this test
+      auto state = ompl_state->as<ompl::base::ConstrainedStateSpace::StateType>()
+                       ->getState()
+                       ->as<ompl_interface::ConstrainedPlanningStateSpace::StateType>();
+
+      // ompl_state.reals() throws a segmentation fault for this state type
+      // use a more involved conversion to std::vector for logging
+      ROS_INFO_STREAM(std::vector<double>(state->values, state->values + joint_model_group_->getVariableCount()));
+    }
+
+    // the given state is known to be in self-collision, we check it here
+    bool result = checker->isValid(ompl_state.get());
+    EXPECT_FALSE(result);
+
+    // do the same for the version with distance
+    double distance{ 0.0 };
+    result = checker->isValid(ompl_state.get(), distance);
+    EXPECT_FALSE(result);
+
+    // but it should respect the joint limits
+    bool result_2 = robot_state_->satisfiesBounds();
+    EXPECT_TRUE(result_2);
+  }
+
+  // /***************************************************************************
+  //  * END Test implementation
+  //  * ************************************************************************/
+
+protected:
+  void SetUp() override
+  {
+    // setup all the input we need to create a StateValidityChecker
+    setupStateSpace();
+    setupPlanningContext();
+  };
+
+  void TearDown() override
+  {
+  }
+
+  void setupStateSpace()
+  {
+    ompl_interface::ModelBasedStateSpaceSpecification space_spec(robot_model_, group_name_);
+    state_space_ = std::make_shared<ompl_interface::ConstrainedPlanningStateSpace>(space_spec);
+    state_space_->computeLocations();  // this gets called in the state space factory normally
+
+    assert(state_space_ != nullptr);
+
+    auto dummy_constraint = std::make_shared<DummyConstraint>(num_dofs_);
+    constrained_state_space_ = std::make_shared<ompl::base::ProjectedStateSpace>(state_space_, dummy_constraint);
+
+    assert(constrained_state_space_ != nullptr);
+  }
+
+  void setupPlanningContext()
+  {
+    ASSERT_NE(state_space_, nullptr) << "Initialize state space before creating the planning context.";
+    planning_context_spec_.state_space_ = state_space_;
+
+    // IMPORTANT, we need to create the simple setup with  a ConstrainedSpaceInformation object,
+    // not with the state space, as this will allocate the wrong type of SpaceInformation
+    auto csi = std::make_shared<ompl::base::ConstrainedSpaceInformation>(constrained_state_space_);
+    planning_context_spec_.ompl_simple_setup_ = std::make_shared<ompl::geometric::SimpleSetup>(csi);
+
+    // check if the we succeeded in the comment above
+    auto si = planning_context_spec_.ompl_simple_setup_->getSpaceInformation();
+    auto si_constrained = dynamic_cast<ompl::base::ConstrainedSpaceInformation*>(si.get());
+    assert(si_constrained != nullptr);
```

Comment:
Use one of the gtest macros for this.

```suggestion
    ASSERT_NE(si_constrained, nullptr);
```

Reply:
done